src/coreclr/vm/stubhelpers.cpp

```diff
@@ -476,17 +476,37 @@ FCIMPL1(void*, StubHelpers::GetDelegateTarget, DelegateObject *pThisUNSAFE)
 }
 FCIMPLEND
 
+#include <optsmallperfcritical.h>
+FCIMPL2(FC_BOOL_RET, StubHelpers::TryGetStringTrailByte, StringObject* thisRefUNSAFE, UINT8 *pbData)
```

Comment:
Can we do this FCall as a `SuppressGCTransition` QCall, or is the perf hit too much for that case?

Reply:
Yes, FCall with SuppressGCTransition would be slower.

I am not worried about FCalls without HMF for this round. I just moved this one over from String to StubHelpers since it does not belong to String. The implementation in the original place was [compiled for speed](https://github.com/dotnet/runtime/pull/95670/files#diff-9baf1cf40192154aa8972811baf859ed33a78d172ce238c7eebf6418d968b831L26-L32) so the perf of this helper can matter somewhere. I prefer to keep it as is.